adapters/alkimi/alkimi_test.go

```diff
@@ -0,0 +1,181 @@
+package alkimi
+
+import (
+	"encoding/json"
+	"github.com/prebid/openrtb/v19/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"testing"
+
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	alkimiTestEndpoint = "https://exchange.alkimi-onboarding.com/server/bid"
+)
+
+func TestEndpointEmpty(t *testing.T) {
+	_, buildErr := Builder(openrtb_ext.BidderAlkimi, config.Adapter{
+		Endpoint: ""}, config.Server{ExternalUrl: "http://hosturl.com", GvlID: 1, DataCenter: "2"})
+	assert.Error(t, buildErr)
+}
+
+func TestEndpointMalformed(t *testing.T) {
+	_, buildErr := Builder(openrtb_ext.BidderAlkimi, config.Adapter{
+		Endpoint: " http://leading.space.is.invalid"}, config.Server{ExternalUrl: "http://hosturl.com", GvlID: 1, DataCenter: "2"})
+	assert.Error(t, buildErr)
+}
+
+func TestBuilder(t *testing.T) {
+	bidder, buildErr := buildBidder()
+	if buildErr != nil {
+		t.Fatalf("Failed to build bidder: %v", buildErr)
+	}
+	assert.NotNil(t, bidder)
+}
+
+func TestMakeRequests(t *testing.T) {
```

Comment:
@kalidas-alkimi  recommended to use JSON framework tests. Should add exemplary and supplemental tests.

Refer following as example,
-  https://github.com/prebid/prebid-server/blob/61a209ae876cd4119474029914fd8d1684f98a8c/adapters/appnexus/appnexustest/exemplary 
- https://github.com/prebid/prebid-server/blob/61a209ae876cd4119474029914fd8d1684f98a8c/adapters/appnexus/appnexustest/supplemental

Reply:
fixed

Reply:
> fixed

@kalidas-alkimi  could you link the added SON framework tests

Reply:
> I hope this is also a recommendation, we will update it in future.

PBS-GO expects to add json framework tests whenever new adapter gets added

Reply:
We don't fully understand, JSON tests are required or not? Should we re-write our unit tests or not?

Reply:
> > I hope this is also a recommendation, we will update it in future.
> 
> PBS-GO expects to add json framework tests whenever new adapter gets added

@pro-nsk  JSON tests are required

Reply:
> @onkarvhanumante Added JSON tests

@pro-nsk  JSON test is expected to cover code path of  MakeRequests and MakeBids. Now that JSON tests are added, could remove following tests.

`TestMakeRequests, TestMakeBidsShouldReturnErrorIfResponseBodyContainsIncorrectImp, TestMakeBidsShouldReturnErrorIfResponseBodyContainsIncorrectImp,
TestMakeBidsShouldReturnEmptyListIfBidResponseIsNull,
TestMakeBidsShouldReturnEmptyListIfBidResponseIsError,
TestMakeBidsShouldReturnBidWithResolvedMacros,
TestMakeBidsShouldReturnErrorIfResponseBodyContainsIncorrectImp,
TestMakeBidsShouldReturnEmptyListIfBidResponseIsNull,
TestMakeBidsShouldReturnEmptyListIfBidResponseIsError,
TestMakeBidsShouldReturnBidWithResolvedMacros,
TestMakeBidsShouldReturnBidForAllTypes
```

Reply:
@onkarvhanumante Fixed